### PR TITLE
fix(modal): adds button spacing on danger modal

### PR DESCRIPTION
**Pull request by davidmenendez**
Resolves https://github.com/carbon-design-system/carbon-components/issues/624

### CODE REVIEW

**Review by joshblack — APPROVED** [review]

File: src/components/button/_button.scss [hunk 1]
```diff
@@ -176,7 +176,8 @@
     letter-spacing: 0;
   }
 
-  .#{$prefix}--btn--secondary + .#{$prefix}--btn--primary {
+  .#{$prefix}--btn--secondary + .#{$prefix}--btn--primary,
+  .#{$prefix}--btn--tertiary + .#{$prefix}--btn--danger--primary {
```
Comment (asudoh) [inline src/components/button/_button.scss:180]: Wondering if this line should simply be `.#{$prefix}--btn + .#{$prefix}—btn`, probably in next major version? CC @carbon-design-system/ui

Comment (alisonjoseph) [reply src/components/button/_button.scss:180]: I think the intention here was that there should never be more than one primary button? I'm not completely sure though.

Comment (tw15egan) [reply src/components/button/_button.scss:180]: @alisonjoseph @asudoh ☝️ I'm pretty sure that's the reasoning when @hellobrian was working on this.

**Review by tw15egan — APPROVED** [review]

File: src/components/button/_button.scss (hunk 1 above)
Comment (marijohannessen) [reply src/components/button/_button.scss:180]: I think so too 👍

**Review by marijohannessen — APPROVED** [review]

**Pull request merged**